File: python/0038.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        def cnt(s,ind):
            cnt=0
            i=ind
            while i<len(s) and s[ind]==s[i]:
                cnt+=1
                i+=1
            return cnt
        s="1"
        for i in range(n-1):
            j=0
            now=""
            while j<len(s):
                sz=cnt(s,j)
                now+=str(sz)+str(s[j])
                j+=sz
            s=now
        return s

File: rust/src/bin/0038.rs
struct Solution;

impl Solution {
    pub fn count_and_say(n: i32) -> String {
        let mut s = "1".to_string();
        for i in 2..=n {
            let mut a = String::new();
            let mut cnt = 0;
            let mut last = 'a';
            for ch in s.chars() {
                if ch == last {
                    cnt += 1;
                } else {
                    if cnt > 0 {
                        a += cnt.to_string().as_str();
                        a.push(last);
                    }
                    cnt = 1;
                    last = ch;
                }
            }
            if cnt > 0 {
                a += cnt.to_string().as_str();
                a.push(last);
            }
            s = a;
            println!("{} {}", i, s);
        }
        s
    }
}

fn main() {
    Solution::count_and_say(10);
}